assets/src/components/selection/event-select/index.js

```diff
@@ -0,0 +1,165 @@
+      selectedEventId,
+      isLoading = true
+    } = this.props;
+    if (isLoading || isEmpty(events)) {
+      return <Placeholder key="placeholder"
+                          icon="calendar"
+                          label={ __('EventSelect') }
+                          >
+        { isLoading
+            ? <Spinner/>
+            : __(
+                'There are no events to select from. You need to create an event first.'
+            )
+        }
+      </Placeholder>
+    }
+
+    return <SelectControl
+            label={ selectLabel }
+            value={ selectedEventId }
+            options={ buildEventOptions(events) }
+            onChange={ (value) => onEventSelect(value) }
+        />;
+  }
+}
+
+/**
+ * @todo some of these proptypes are likely reusable in various place so we may
+ * want to consider extracting them into a separate file/object that can be
+ * included as needed.
+ * @type {{events: *, onEventSelect, selectLabel: *, selectedEventId: *, isLoading: *, attributes: {limit: *, orderBy: *, order: *, showExpired: *, categorySlug: *, month: *}}}
+ */
+EventSelect.propTypes = {
+  events: PropTypes.shape({
+    EVT_name: PropTypes.string.required,
+    EVT_ID: PropTypes.number.required
+  }),
+  onEventSelect: PropTypes.func.required,
+  selectLabel: PropTypes.string,
+  selectedEventId: PropTypes.number,
+  isLoading: PropTypes.bool,
+  attributes: PropTypes.shape({
+    limit: PropTypes.number,
+    orderBy: PropTypes.oneOf(['EVT_name',
+        'EVT_ID',
+        'start_date',
+        'end_date',
+        'ticket_start',
+        'ticket_end'
+    ]),
+    order: PropTypes.oneOf(['asc', 'desc']),
+    showExpired: PropTypes.bool,
+    categorySlug: PropTypes.string,
+    month: PropTypes.month
+  })
+};
+
+EventSelect.defaultProps = {
+  attributes: {
+    limit: 20,
+    orderBy: 'start_date',
+    order: 'desc',
+    showExpired: false,
+  }
+};
+
+/**
+ * Used to map an orderBy string to the actual value used in a REST query from
+ * the context of an event.
+ * @todo this should be moved to a mapper library for various EE Rest Related
+ * things maybe?
+ * @param orderBy
+ * @returns {{start_date: string, end_date: string, ticket_start: string, ticket_end: string}}
+ */
+const mapOrderBy = (orderBy) => {
+  const orderByMap = {
+    start_date: 'Datetime.DTT_EVT_start',
+    end_date: 'Datetime.DTT_EVT_end',
+    ticket_start: 'Datetime.Ticket.TKT_start_date',
+    ticket_end: 'Datetime.Ticket.TKT_end_date'
+  };
+  return isUndefined(orderByMap[orderBy]) ? orderBy : orderByMap;
+};
+
+const whereConditions = ({showExpired, categorySlug, month}) => {
+  let where = [];
+  const GREATER_AND_EQUAL = encodeURIComponent('>=');
+  const LESS_AND_EQUAL = encodeURIComponent('<=');
+  //using .replace('Z', '') on the end of our moment strings because EE currently has a bug with parsing default
+  //moment formatted strings (@see https://events.codebasehq.com/projects/event-espresso/tickets/11368)
```

Comment:
Did you forget to do this replacement?
I don't see this applied anywhere in the code below!?!

Reply:
hmm nope, copy/pasta from code I did elsewhere (the eea-gutenberg add-on), the issue got fixed upstream in EE core and thus its no longer needed.  I'll just remove this comment.